### PR TITLE
refactor(api): generalize ldap

### PR DESCRIPTION
**Pull request by fsamin**
Signed-off-by: Brandon Nason <brandon.nason@gmail.com>

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#13513.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✘  
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#13525.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package and Analyze
  * Docker ✔ 
  * Docs ✔ 
  * Sonar ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘  
  * Standalone tests 1 -   
  * Standalone tests 2 -   
  * Standalone tests 3 -

File: engine/api/authentication/ldap/ldap.go [hunk 1]
```diff
@@ -76,14 +79,23 @@ func (d AuthDriver) CheckSigninRequest(req sdk.AuthConsumerSigninRequest) error
...
 	var password = req["password"]
 
-	if err := d.bind(ctx, bind, password); err != nil {
-		return userInfo, sdk.NewError(sdk.ErrUnauthorized, err)
+	if d.conf.BindDN != "" {
+		if err := d.bindDN(ctx, d.conf.BindDN, d.conf.BindPW); err != nil {
+			return userInfo, sdk.NewError(sdk.ErrUnauthorized, err)
+		}
+	} else {
+		userDN := strings.Replace(d.conf.UserDN, "{0}", ldap.EscapeFilter(user), 1)
```
Comment (fsamin) [inline engine/api/authentication/ldap/ldap.go:90]: `d.conf.UserDN` doesn't seems to be initialized

Comment (bnason) [reply engine/api/authentication/ldap/ldap.go:90]: I've pushed a new commit that should fix this https://github.com/bnason/cds/commit/e1925c700f9370a78a2d4ead8772cec6ed89f911

**Comment by bnason** [comment]
@fsamin not sure if you saw my comment on the file, but I pushed a fix. https://github.com/ovh/cds/pull/5237#discussion_r436818248

**Pull request closed without merging**